src/pages/Main.js

```diff
@@ -1,66 +1,137 @@
+          alert('이미 출근하셨습니다.');
+        }
+        if (Object.keys(errorMessage.result).length > 0) {
+          alert(errorMessage.result.comment);
+        }
+        if (errorMessage.message === 'LOCATION_ERROR') {
+          alert('위코드에 오시긴 하셨나요?');
+        }
+        history.push('/main');
+      });
   };
 
-  const changeStartText = () => {
-    setStartDescription(
-      `은 ${getTime()}시 ${new Date().getMinutes()}분에 시작하셨습니다.`
-    );
+  const checkEnd = e => {
+    fetch('http://10.58.6.89:8000/records/2')
+      .then(res => res.json())
+      .then(errorMessage => {
+        if (errorMessage.message === 'NEED_TO_RECORD_STARTTIME_ERROR') {
+          alert('출석부터 누르세요');
+        }
+        if (errorMessage.message === 'ALREADY_RECORD_ERROR') {
+          alert('이미 퇴근했습니다.');
+        }
+        if (errorMessage.message === 'LOCATION_ERROR') {
+          alert('위코드에 계시긴 하나요?');
+        }
+        history.push('/main');
+      });
   };
 
   return (
     <Container>
       <TimeSection>
         <TimeDescription>
-          지금은 {new Date().getMonth() + 1}월 {new Date().getDate()}일
-          {day[new Date().getDay()]}요일
+          {`지금은 ${new Date().getMonth() + 1}월 ${new Date().getDate()}일 ${
+            day[new Date().getDay()]
+          }요일`}
         </TimeDescription>
         <TimeDescription>
-          {getTime()}시 {getMinutes}분 입니다.
+          {`${getTime(time.hour)}시 ${
+            time.minutes !== 0 ? `${time.minutes}분` : ''
+          }입니다.`}
         </TimeDescription>
       </TimeSection>
       <StartSection>
-        <StudentName>이다슬님</StudentName>
-        <StartTime>{startDescription}</StartTime>
+        {!userInfo.isOn ? (
+          <StartTime>오늘도 상쾌하게 시작해볼까요?</StartTime>
+        ) : (
+          <>
+            <StudentName>{userInfo.name}</StudentName>
+            <StartTime>{`님은 ${startTimeArr[0]}시 ${startTimeArr[1]}분에 시작하셨습니다.`}</StartTime>
+          </>
+        )}
       </StartSection>
       <ButtonAnimationSection>
         <ButtonSection>
-          <Button onClick={changeStartText}>START</Button>
-          <Button>STOP</Button>
+          <Button onClick={checkStart}>START</Button>
+          <Button onClick={checkEnd}>STOP</Button>
         </ButtonSection>
         <FireAnimationSection>
-          <FirewoodImg
-            alt="firewoodimg"
-            src="/images/firewood.png"
-          ></FirewoodImg>
-          <FireGif alt="firegif" src="/images/fire.gif"></FireGif>
+          <FirewoodImg alt="firewoodimg" src="/images/firewood.png" />
+          {userInfo.isOn && <FireGif alt="firegif" src="/images/fire.gif" />}
         </FireAnimationSection>
       </ButtonAnimationSection>
+
+      {userInfo.normalAttendance && (
+        <Modal height="300px">
+          <SendTimeModal can={userInfo.normalAttendance} />
+        </Modal>
+      )}
     </Container>
   );
-};
+}
 
-export default Main;
+const day = ['일', '월', '화', '수', '목', '금', '토'];
```

Comment:
```suggestion
const WEEK = ['일', '월', '화', '수', '목', '금', '토'];
```
변하지 않는 상수라면, 이건 어떨까요??

Reply:
@daseuls 다슬님~ 여기요~~